l10n_fr_hr_rup/models/hr_contract.py

```diff
@@ -7,14 +7,31 @@ class HrContract(models.Model):
     _inherit = "hr.contract"
     _order = "date_start,name"
 
-    pcs_id = fields.Many2one("hr.employee.pcs", "PCS")
     employer_address_id = fields.Many2one("res.partner", "Addresse de l'employeur")
-    qualification = fields.Char("Qualification")
-    work_location = fields.Char("Localisation du bureau")
+    pcs_id = fields.Many2one(
+        "hr.employee.pcs", "PCS", compute="_compute_pcs_id", store=True, readonly=False
+    )
+    qualification = fields.Char(
+        "Qualification", compute="_compute_qualification", store=True, readonly=False
+    )
+    work_location = fields.Char(
+        "Localisation du bureau",
+        compute="_compute_work_location",
+        store=True,
+        readonly=False,
+    )
 
-    @api.onchange("employee_id")
-    def _onchange_employee_id(self):
-        super()._onchange_employee_id()
-        self.pcs_id = self.employee_id.pcs_id
-        self.qualification = self.employee_id.qualification
-        self.work_location = self.employee_id.work_location
+    @api.depends("employee_id")
+    def _compute_pcs_id(self):
+        for contract_id in self:
+            contract_id.pcs_id = self.employee_id.pcs_id
```

Comment:
```suggestion
        for contract_id in self:
            contract_id.pcs_id = contract_id.employee_id.pcs_id
```

Also change the others please.
Otherwise you get an error when recomputing multiple records




Also, as a non blocking suggestion, I prefer naming the single record `rec` or `record`, as it's pretty common to find it like this and it's a "standard" that applies to any model

```suggestion
        for rec in self:
            rec.pcs_id = rec.employee_id.pcs_id
```

Reply:
Ops, sorry for this mistake and thanks again for the advises!